src/rgw/rgw_op.cc

```diff
@@ -5245,7 +5246,7 @@ void RGWCopyObj::execute(optional_yield y)
 	   s->yield);
 
   // send request to notification manager
-  int ret = res->publish_commit(this, s->obj_size, mtime, etag);
+  int ret = res->publish_commit(this, astate->size, mtime, etag);
```

Comment:
hi @yuvalif, Mark identified this commit as the likely cause of new valgind issues, https://tracker.ceph.com/issues/51347

`astate` here is pointing at memory in `std::map<rgw_obj, RGWObjState> RGWObjectCtx::objs_state`, but this gets invalidated in `RGWRados::Object::Write::_do_write_meta()` if we detect a racing write to the head object and have to re-read it

so it's not safe to deference `astate` here; we either need to save the value of `astate->size` into a temporary and use that for `publish_commit()`, or call `get_obj_state()` again after this call to `copy_object()` to see the updated value in case it changed

Reply:
@cbodley what is the difference between `astate->size` and `astate->accounted_size`?

Reply:
@yuvalif accounted_size rounds sizes to some block size iirc (was used in the past for trying to account for actual storage used)

Reply:
@yehudasa thanks.